src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -2029,4 +2031,19 @@ public function __clone()
     {
         $this->query = clone $this->query;
     }
+
+    /**
+     * Wraps the given Closure with a transaction savepoint if needed.
+     *
+     * @template TModelValue
+     *
+     * @param  \Closure(): TModelValue  $scope
+     * @return TModelValue
+     */
+    public function withSavepointIfNeeded(Closure $scope): mixed
```

Comment:
Not sure about this. I thought about just wrapping the insert part in a `DB::transaction(fn)`, which not sure what y'all would think of that. I initially had it as `protected`, but I needed the same logic in the relationship part of the feature.

Let me know what y'all think.

Reply:
For MySQL, this is not necessary; Postgres just needs it. However, it is not known what SQLServer and SQLite do.

Reply:
Yeah, I figured it wouldn't be a problem to wrap in savepoint transactions for the other drivers as well. It doesn't seem to hurt.